### PR TITLE
Docs index: Group tasks by tags

### PR DESCRIPTION
**Pull request by vfonic**
I came to `docs/README.md` and was overwhelmed and lost in the huuuuuge task list.
I figured there's already this whole powerful and well-organized system for grouping tasks by tags, so why not use it in the docs index?
I put tasks grouped by tags below the gigantic alphabetical list of tasks. Perhaps we should move it above instead?

Thanks!

You can preview the new `docs/README.md` here: https://github.com/vfonic/mechanic-tasks/blob/docs/group-tasks-by-tags/docs/README.md

### CODE REVIEW

**Comment by vfonic** [comment]
Something seems to be broken with CLA bot. Also can anyone explain to me what's the purpose of the CLA bot? IMO it's just a waste of time, but maybe there's a reason behind it.

File: lib/util.js
Comment (isaacbowen) [inline lib/util.js:1]: please nix the code style changes, leaving only the changes that are material to the purpose of this pull request.

(I'm wiiide open to having prettier or something in this repo, but I want to leave changes of that sort to their own pr.)

**Comment by isaacbowen** [comment]
hey @vfonic! thanks for this! and yes, the bot is somewhat unwell. :) it might still work though: try leaving a comment in this pr consisting solely of this text, unformatted:

> I have read the CLA Document and I hereby sign the CLA

the purpose for this is to have a written record of each contributor's IP release, making for uncomplicated [licensing of the whole](https://github.com/lightward/mechanic-tasks/blob/main/LICENSE).

**Comment by vfonic** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/util.js
Comment (isaacbowen) [reply lib/util.js:1]: oh my god I just noticed that you _did_ keep the prettier changes in their own commit. fantastic!

lightward keeps each pull request tightly scoped to its stated purpose; can you push [7f821e1](https://github.com/lightward/mechanic-tasks/pull/254/commits/7f821e1021ca1580734090f79043e9c1f1e98603) into its own branch, and make its own pr?

**Comment by vfonic** [comment]
I just updated the PR with any formatting changes removed.
